### PR TITLE
Populate CPU convolution and print out results

### PR DESCRIPTION
**Pull request by yiqian1**
- Add an mlir-miopen-driver option -prc to populate CPU convolution without invoking gpu convolution and print out results.
- Add unit tests to verify CPU convolution results for small tensors. Unit tests cover 3 directions, 2 layouts, and non-default paddings, dilations, and strides.

The new function populateCpuConvolutionLogic() and exiting function populateValidationLogic() have redundant code. I will simplify them in my next PR.

### CODE REVIEW

File: mlir/test/mlir-miopen-driver/cpu_convolution.mlir [hunk 1]
```diff
@@ -0,0 +1,142 @@
+//RUN: mlir-miopen-driver -prc -fil_layout=kcyx -in_layout=nchw -out_layout=nkhw -batchsize=1 -in_channels=2 -out_channels=2 -in_h=8 -in_w=8 -fil_h=3 -fil_w=3   --dilation_h=1 --dilation_w=1 --padding_h=0 --padding_w=0 --conv_stride_h=1 --conv_stride_w=1 -p=false -c| mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=CHECK1
```
Comment (whchung) [inline mlir/test/mlir-miopen-driver/cpu_convolution.mlir]: @yiqian1 please add comment to show the traits of tests conducted in this unit test. for example:
- 3x3, without padding

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp [hunk 2]
```diff
@@ -1553,6 +1555,154 @@ static LogicalResult populateValidationLogic(
...
+                            SmallVector<int64_t, 4> &outputDimension) {
+  // Construct main function.
+  auto func = FuncOp::create(builder.getUnknownLoc(), "main",
+                             builder.getFunctionType({}, {}));
+  module.push_back(func);
+
+  // Construct a new Block.
+  Block *block = func.addEntryBlock();
+
+  auto floatType = builder.getF32Type();
```
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp:1573]: @yiqian1 please document that `populateCpuConvolutionLogic` ONLY produces convolution logic for f32 type.

**Review by whchung — CHANGES_REQUESTED** [review]
Please enhance the test to check all permutations below:

- data type: f32
- layout : nchw / nhwc
- direction: forward / backward data / backward filter
- filter size: 1x1 / 3x3
- padding : 0 / 1
- dilation : 1
- stride : 1 / 2

There should be 48 cases.

**Comment by yiqian1** [comment]
dilation=0 and stride=0 make no sense. I'll test dilation = 1 , 2 and stride = 1, 2. Or do you think we should cover more cases, like stride = 3? @whchung

**Comment by whchung** [comment]
indeed dilation = 0 and stride = 0 makes no sense. for stride let's keep it to 1 / 2. we don't have configs which does stride = 3 as of yet.

**Review by whchung — APPROVED** [review]

**Pull request merged**